### PR TITLE
added figures, results section, commented parts

### PR DESCRIPTION
**Pull request by HamidHeyde**

### CODE REVIEW

File: paper/sections/materials.tex [hunk 1]
```diff
@@ -9,16 +9,38 @@ \section{Materials and Methods}
...
+10,780 transporter, carrier, and channel proteins that were well characterized at the protein level and had clear substrate 
+annotations from the SwissProt(release 2013-03)~\cite{boeckmann2003swiss}. They then removed the transporters 
+with more than two substrate specificities, sequences with biological function annotations based solely on sequence 
+similarity, and sequences with greater than 70\% similarity. The final dataset contains 660 non-transporters and 
+900 transporters being divided into following 7 substrate classes: amino acid, anion, cation, electron, protein/mRNA, 
+sugar, and other.\\
+These $1,560$ sequences were then divided into two datasets: 1) Main dataset for the total of $1,380$ sequences 
+and 2) Independent Dataset which contains $180$ sequences. The work in~\cite{mishra2014prediction} reports on results of 
+all the developed models on the main dataset and the results from the best model (AAIndex+PSSM) on the independent set. 
+Through this study, we used the main dataset and experimented on all the models being developed on that set.
```
Comment (gkiar) [inline paper/sections/materials.tex:37]: @HamidHeyde  - thanks for clarifying. While they may use the terms "main" and "independent," this is very standardly referred to as train/test set, and validation set. The validation set is deliberately left out of any exploration and is used once you've picked your model. So, while we're evaluating performance on our train/test sets, the fact that we never compare performance on the validation set to me feels like a large piece that we are missing: our models may have higher accuracy/F1/precision/recall/you-name-it, but they could also generalize more poorly out-of-sample; it is impossible to know the answer without using a validation set, as well.

Can you please create the independent dataset and evaluate our models (or at least the "best" one(s)) using it?

@glatard , what are your thoughts on this?

File: paper/sections/materials.tex [hunk 2]
```diff
@@ -29,16 +51,26 @@ \subsection{Dataset}
 percentage of all standard amino acids present within a protein, as defined by~\cite{gromiha2010protein}. Also known
 as Monopeptide Composition (MPC).
 \item \textbf{Dipeptide Composition (DPC)}: a feature vector of $400$ values ranging from $0$ -- $100$ indicating the
-percentage of all possible ordered dipeptide pairs present within a protein, as defined by~\cite{gromiha2010protein}.
-\item \textbf{Physico-Chemical Composition (PHC)}: a feature vector of $11$ present/absent binary values corresponding
-to physico-chemical residue classes, including: Aliphatic, Neutral, Aromatic, Hydrophobic, Charged, Positively charged,
+percentage of all possible amino acid pairs present within a protein, as defined by~\cite{gromiha2010protein}.
```
Comment (gkiar) [inline paper/sections/materials.tex:54]:
```suggestion
percentage of all possible ordered amino acid pairs present within a protein, as defined by~\cite{gromiha2010protein}.
```

The fact that these pairs are ordered is important, as it changes the length from 20c2 (190) to 20^2 (400)

File: paper/sections/materials.tex [hunk 3]
```diff
@@ -29,16 +51,26 @@ \subsection{Dataset}
...
+\item \textbf{Physico-Chemical Composition (PHC)}: a feature vector of $11$ values corresponding
+to percentage composition of physico-chemical residue classes, including: Aliphatic, Neutral, Aromatic, Hydrophobic, Charged, Positively charged,
 Negatively charged, Polar, Small, Large, and Tiny. 
 \item \textbf{Biochemical Composition (AAindex)}: a feature vector of $49$ physical, chemical, energetic, and
 conformational amino acid properties which have been averaged across all amino acids present within the protein.
-\item \textbf{Position-specific scoring matrix (PSSM) profile}: a feature vector of $400$ min-maxed scaled probabilities
-for the occurrence of each amino acid residue and position combination within the sequence. \GK{Hamid, please verify this sentence}.
+\item \textbf{Position-specific scoring matrix (PSSM) profile}: a feature vector of $400$ min-maxed scaled values 
+being generated by summing up all the rows (scores) in the PSSM profile of a sequence that correspond to the same amino acid.
```
Comment (gkiar) [inline paper/sections/materials.tex:61]: @HamidHeyde this edit doesn't tell us the meaning of the rows/scores in the PSSM profile. Aren't they probabilities of finding given sequences at given locations? This sentence should make it explicit what the "score" being normalized is, rather than recursively referencing the PSSM profile for its definition.

File: paper/sections/materials.tex [hunk 4]
```diff
@@ -51,13 +83,15 @@ \subsection{Model Flexibility}
 \item the sorting and balancing of samples within the dataset,
 \item the selected SVM hyperparameters, gamma and cost,
 \item the uniformity (or possible lack of) SVM hyperparameters across binary classifiers,
-\item the aggregation technique applied to binary classifiers, and
-\item the prediction method for the final labels.
+\item the aggregation technique applied to binary classifiers,
+\item the prediction method for the final labels, and\\
+\Hamid{for evalueation metrics comment below:}
+{\item the evaluation metrics averaging technique (macro vs micro).}
```
Comment (gkiar) [inline paper/sections/materials.tex:89]: @HamidHeyde it looks like you added another bullet here, but didn't actually add any number in the text below which references this bullets. But more importantly, I don't think that this new bullet should actually exist as the "averaging technique" you added IS the "aggregation technique" two bullets above. See the first paragraph of "Performance Evaluation," below. The change on this and the preceding three two lines should be reverted.

File: paper/sections/materials.tex (hunk 1 above)
Comment (glatard) [reply paper/sections/materials.tex:37]: Hey @gkiar, thanks for pointing this out! As discussed, I agree that we should evaluate our models on the independent set (180 sequences), in particular:
* Measure how far the model closest to the paper model is, in the independent set (replication aim)
* Measure what is the variability across all our models, in the independent set (methods flexibility aim)
@HamidHeyde, do you think you could do that?

Comment (HamidHeyde) [reply paper/sections/materials.tex:37]: Hi @glatard, @gkiar.
Thanks for the review @gkiar,

So, if i'm right, you guys want me to try all the 156 models on the validation set and report the results? (Table 1,2,3 for validation set as well?)

File: paper/sections/materials.tex (hunk 2 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:54]: So, if there are 400 features, then that mean, it was not ordered? because there are 400 numerical values for each sequence in this regard.

Comment (gkiar) [reply paper/sections/materials.tex:54]: 400 values means that it **is** ordered: position 1 (20) x position 2 (20) = 400. If order didn't matter, then it would just be 20c2 = 20! / (18!) (2!) = 20*19/2 = 190.

File: paper/sections/materials.tex (hunk 3 above)
Comment (gkiar) [reply paper/sections/materials.tex:61]: Thanks! Try to keep it to one sentence, somewhere like what I had written just factually correct :)

File: paper/sections/materials.tex (hunk 4 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:89]: I just added a bullet point trying to take care of the comment underneath that list. i should have replaced the last bullet point (6) with this one. Because 5 and 6 are kind of the same thing. 👍

File: paper/sections/materials.tex (hunk 1 above)
Comment (gkiar) [reply paper/sections/materials.tex:37]: Yep, exactly! Just grab the extra 180 sequences from the database, matching the distributions in the original paper, and then evaluate **every** already-fit model on this new collection. :)

File: paper/sections/materials.tex (hunk 2 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:54]: Ok 👍

File: paper/sections/materials.tex (hunk 1 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:37]: Ok, it may take some time, but no problem 👍

File: paper/sections/materials.tex (hunk 3 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:61]: i'll try my best to keep it short

File: paper/sections/materials.tex (hunk 1 above)
Comment (gkiar) [reply paper/sections/materials.tex:37]: Perfect. I'll merge this PR and start reviewing your methods once the other (text based) edits are done, and we can merge the changes as a result of this afterwards. Sound good?

Comment (HamidHeyde) [reply paper/sections/materials.tex:37]: Great 👍

File: paper/sections/materials.tex [hunk 5]
```diff
@@ -83,10 +86,9 @@ \subsection{Model Flexibility}
 \item the sorting and balancing of samples within the dataset,
 \item the selected SVM hyperparameters, gamma and cost,
 \item the uniformity (or possible lack of) SVM hyperparameters across binary classifiers,
-\item the aggregation technique applied to binary classifiers,
-\item the prediction method for the final labels, and\\
+\item the aggregation technique applied to binary classifiers, and \\
 \Hamid{for evalueation metrics comment below:}
-{\item the evaluation metrics averaging technique (macro vs micro).}
+{\item the averaging technique for calculation of final performance metrics of a model (macro vs micro).}
```
Comment (gkiar) [inline paper/sections/materials.tex:91]: @HamidHeyde please revert the changes on the current lines 89 and 91. The changes break the order of the list and the downstream paragraph. The bullets aren't the same -- one is *when* the aggregation takes place (micro vs macro), the other is *how* the aggregation is done (max prob, weighted-mean, or unweighted mean).

File: paper/sections/materials.tex [hunk 6]
```diff
@@ -51,14 +51,17 @@ \subsection{Dataset}
...
 Negatively charged, Polar, Small, Large, and Tiny. 
 \item \textbf{Biochemical Composition (AAindex)}: a feature vector of $49$ physical, chemical, energetic, and
 conformational amino acid properties which have been averaged across all amino acids present within the protein.
-\item \textbf{Position-specific scoring matrix (PSSM) profile}: a feature vector of $400$ min-maxed scaled values 
-being generated by summing up all the rows (scores) in the PSSM profile of a sequence that correspond to the same amino acid.
+\item \textbf{Position-specific scoring matrix (PSSM) profile}: A Position-Specific Scoring Matrix (PSSM), 
+is a scoring matrix in which amino acid substitution scores are given separately for each position in 
+a protein multiple sequence alignment. This feature is a min-maxed scaled vector of $400$ values being 
+generated by summing up all the rows of the PSSM of a sequence corresponding to the same amino acid. 
+The PSSM profile was generated by running PSI-BLAST against SwissProt database with the BLOSUM62 matrix.
```
Comment (gkiar) [inline paper/sections/materials.tex:64]:
```suggestion
\item \textbf{Position-specific scoring matrix (PSSM) profile}: a feature vector of $400$ sequence
likelihoods aggregated across min-max scaled probabilities for each ordered amino acid pair within
proteins in the SwissProt database.
```

File: paper/sections/materials.tex (hunk 5 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:91]: @gkiar: You mean i need to change the order? so, micro vs macro => 5 and aggregation technique => 6  or revert it back to what it was before?

Comment (gkiar) [reply paper/sections/materials.tex:91]: I mean to exactly as they were written before this PR, yes. Thanks!

File: paper/sections/materials.tex (hunk 6 above)
Comment (gkiar) [reply paper/sections/materials.tex:64]: @HamidHeyde this is the other change to look at right now - I think you'll just need to hit "approve" and we're all set

Comment (HamidHeyde) [reply paper/sections/materials.tex]: @gkiar: You guys are more experienced in this regard, if you think these 2 are equal, please go ahead and change it, cause i couldn't make it shorter than those 3 sentences.

File: paper/sections/materials.tex (hunk 5 above)
Comment (HamidHeyde) [reply paper/sections/materials.tex:89]: sure, no problem for me. 
Sorry @gkiar, I'm a bit confused, this way of communication sometimes gives the wrong idea. 
Correct me if i'm wrong, but before it was (5)=>aggregation technique and (6)=>  final label prediction method? and since these 2  (to my understanding) are referring to the same things, then wasn't that why Tristan left that comment (micro vs macro) and i had to add a Bullet Point for that? 

So, instead of aggregation technique(=>5) shouldn't we say, performance metrics averaging technique? since you have already mentioned that point in the "2.3.3-Performance Evaluation" (addr:5)

So, if that is not the case then what i should i write in regards to Tristan's comment? Should i add a new bullet point? or it has been already taken care of?

Comment (gkiar) [reply paper/sections/materials.tex]: Hey @HamidHeyde - I agree, looking at partial diffs does indeed get hard to track and follow!

I believe that @glatard 's comment was covered in the paragraph below and under bullet 5, as we said, because we explicitly link it to the macro/micro approach, there. I'll add a note to suggest that we perhaps adjust the wording of the 5th bullet to make that more clear, but the sequence and content should otherwise be unchanged. Thanks for reverting and the other changes!

Comment (HamidHeyde) [reply paper/sections/materials.tex]: Thanks @gkiar
